mongodb/src/main/java/com/yahoo/ycsb/db/AsyncMongoDbClient.java

```diff
@@ -0,0 +1,477 @@
+ * http://www.apache.org/licenses/LICENSE-2.0                                                                                                                                      
+ *                                                                                                                                                                                 
+ * Unless required by applicable law or agreed to in writing, software                                                                                                             
+ * distributed under the License is distributed on an "AS IS" BASIS,                                                                                                               
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or                                                                                                                 
+ * implied. See the License for the specific language governing                                                                                                                    
+ * permissions and limitations under the License. See accompanying                                                                                                                 
+ * LICENSE file.                                                                                                                                                                   
+ */
+package com.yahoo.ycsb.db;
+
+import static com.allanbank.mongodb.builder.QueryBuilder.where;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.allanbank.mongodb.Durability;
+import com.allanbank.mongodb.LockType;
+import com.allanbank.mongodb.MongoClient;
+import com.allanbank.mongodb.MongoClientConfiguration;
+import com.allanbank.mongodb.MongoCollection;
+import com.allanbank.mongodb.MongoDatabase;
+import com.allanbank.mongodb.MongoDbUri;
+import com.allanbank.mongodb.MongoFactory;
+import com.allanbank.mongodb.MongoIterator;
+import com.allanbank.mongodb.ReadPreference;
+import com.allanbank.mongodb.bson.Document;
+import com.allanbank.mongodb.bson.DocumentAssignable;
+import com.allanbank.mongodb.bson.Element;
+import com.allanbank.mongodb.bson.ElementType;
+import com.allanbank.mongodb.bson.builder.BuilderFactory;
+import com.allanbank.mongodb.bson.builder.DocumentBuilder;
+import com.allanbank.mongodb.bson.element.BinaryElement;
+import com.allanbank.mongodb.builder.Find;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+
+/**
+ * MongoDB asynchronous client for the YCSB framework.
+ * <p>
+ * See the README for details on configuration the client and driver.
+ * </p>
+ * 
+ * @author rjm
+ * @copyright 2013-2014, Allanbank Consulting, Inc., All Rights Reserved
```

Comment:
the project relies on license headers at the top of files and not copyright javadocs.